### PR TITLE
PHP: Fixing a few bugs with phpDoc

### PR DESCRIPTION
**Pull request by troizet**
- Fix bug where phpdoc for a function or method is generated in the wrong block.

Example:
```php
class ClassExample
{

    public function method1()
    {
        
    }

    private function method2($params)
    {
        return false;
    }
}
```

Before:

https://github.com/apache/netbeans/assets/9607501/f2f8b6f6-8d39-4174-83f4-4e8ffe3eb19b

After:

https://github.com/apache/netbeans/assets/9607501/6548b366-6ccc-406f-a01c-ec61219c7a99

- Fix bug where phpdoc from another block is used to display documentation for a function or method.
Example:
```php
class ClassExample
{

    public function method1()
    {
        /**
         *
         * @param string $params
         * @return int
         */
    }

    private function method2(int $params)
    {
        return 'string';
    }
}
```

Before:
phpdoc_before

After:
phpdoc_after

### CODE REVIEW

**Review by tmysik — APPROVED** [review]
Looks harmless to me but please, wait for the approval from @junichi11. Thanks a lot for the unit tests!

File: php/php.editor/src/org/netbeans/modules/php/editor/parser/api/Utils.java
Comment (junichi11) [inline php/php.editor/src/org/netbeans/modules/php/editor/parser/api/Utils.java:83]: Does `c` mean "comment" or "check"?
I prefer another name to `cnode`.

Comment (troizet) [reply php/php.editor/src/org/netbeans/modules/php/editor/parser/api/Utils.java]: Renamed to `nodeAtStartOffset`.

Comment (junichi11) [reply php/php.editor/src/org/netbeans/modules/php/editor/parser/api/Utils.java]: Nice.

**Review by junichi11 — APPROVED** [review]
Thank you for your work.

**Pull request merged**
